src/backend/cuda/texture.hpp

```diff
@@ -0,0 +1,66 @@
+/*******************************************************
+ * Copyright (c) 2020, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <Array.hpp>
+#include <err_cuda.hpp>
+
+#include <type_traits>
+
+namespace cuda {
+
+template<typename T>
+class LookupTable1D {
+   public:
+    LookupTable1D()                          = delete;
+    LookupTable1D(const LookupTable1D& arg)  = delete;
```

Comment:
It makes sense to have a copy constructor for this right? You can copy a texture if you need it.

Reply:
Then we have to take care of how texture object is copied also. Just handle copy won't work. Move operation wont have issues though. We don't need neither for these use cases.